zipkin-server/src/main/java/zipkin2/server/internal/brave/TracingConfiguration.java

```diff
@@ -51,9 +51,16 @@
 @EnableConfigurationProperties(SelfTracingProperties.class)
 @ConditionalOnSelfTracing
 public class TracingConfiguration {
+  static volatile Thread reporterThread;
```

Comment:
Also realized it might be better to have a `TracedCall` factory instead of calling the constructor directly.

Reply:
generally I feel both approaches are a bit crap from an abstraction pov, but I like your first workaround better as it feels less dodgy. Ex traced call shouldn't need special knowledge anyway. the main issue is around instrumentation policy on the edge case of self-tracing inside the tracer though.. and so yeah.. anyway this is good input. I'm hoping moving the sampler around can help.

Reply:
PS both of these are good suggestions.. the main problem is that the call makes a downstream call, and that would still hit a path that trips the request context based implementation (even if not sampled). Ex there's currently no thread-local means to say not just don't sample, but also don't instrument at all. Dancing around the log warning is probably an important pain to feel, even if it is super tempting to just quiet the message  heh.

Reply:
moved the root concern here https://github.com/line/armeria/issues/1964

Reply:
meanwhile one or both of these approaches is a lot of spaghetti drawing attention to something really unfortunate. I like the former is less code less beans less distraction as ideally we'll never have to expose this stuff... ex I don't get insight looking through /actuator/beans and now have 2 more beans just because of this crap logging concern. Meanwhile, I'm just going to kill the log message with a ref to the issue.